plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -153,6 +153,27 @@ internal fun Package.toSpdxPackage(
 
     val resolvedLicenseExpressions = licenseInfoResolver.resolveLicenseInfo(id).filterExcluded()
 
+    val declaredPackageLicenses = resolvedLicenseExpressions.filter(LicenseView.ONLY_DECLARED)
+
+    val matcher = PathLicenseMatcher(LicenseFilePatterns.getInstance())
+    val licensePaths = resolvedLicenseExpressions.flatMap { licenseInfo ->
+        licenseInfo.locations.map { it.location.path }
+    }
+
+    val packageRootPath = vcsProcessed.path.takeIf { type == SpdxPackageType.VCS_PACKAGE }.orEmpty()
+    val applicableLicensePaths = matcher.getApplicableLicenseFilesForDirectories(licensePaths, listOf(packageRootPath))
+    val applicableLicenseFiles = applicableLicensePaths[packageRootPath].orEmpty()
```

Comment:
Inlining `applicableLicensePaths` IMO can avoid confusion with `applicableLicenseFiles`.

Reply:
I prefer to keep this out of the loops implied by `filterTo` and `any`, but I'm open to different names.